crates/biome_js_analyze/src/lint/nursery/use_trim_start_end.rs

```diff
@@ -0,0 +1,292 @@
+                    callee
+                        .as_js_computed_member_expression()?
+                        .r_brack_token()
+                        .ok()?,
+                )
+                .build(),
+            )
+        } else {
+            AnyJsExpression::JsStaticMemberExpression(make::js_static_member_expression(
+                callee_object,
+                callee
+                    .as_js_static_member_expression()?
+                    .operator_token()
+                    .ok()?,
+                AnyJsName::JsName(make::js_name(make::ident(&replaced_member_name))),
+            ))
+        };
+
+        let mut mutation = ctx.root().begin();
+        mutation.replace_node(callee, call_expression);
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Replace "<Emphasis>{state.member_name}</Emphasis>" with "<Emphasis>{replaced_member_name}</Emphasis>"." }
+                .to_owned(),
+            mutation,
+        ))
+    }
+}
+
+fn generate_syntax_token(callee: AnyJsExpression) -> Option<SyntaxToken<JsLanguage>> {
+    let token = if let AnyJsExpression::JsComputedMemberExpression(expression) = callee {
+        let member = expression.member().ok()?;
+        match member {
+            AnyJsExpression::AnyJsLiteralExpression(literal) => literal.value_token().ok(),
+            AnyJsExpression::JsTemplateExpression(element) => {
+                element.elements().into_iter().find_map(|x| {
+                    x.as_js_template_chunk_element()
+                        .and_then(|chunk| chunk.template_chunk_token().ok())
+                })
+            }
+            _ => None,
+        }
+    } else if let AnyJsExpression::JsStaticMemberExpression(expression) = callee {
+        expression.member().ok()?.value_token().ok()
+    } else {
+        None
+    };
+    token
+}
+
+// Handle "'text'" and "\"text\"" and "text" cases
+fn suggested_name(text: &SyntaxToken<JsLanguage>) -> String {
+    let trimmed = text.text_trimmed();
+    let first_char = trimmed.chars().next();
+    let last_char = trimmed.chars().last();
+
+    let is_single_quoted = first_char == Some('\'') && last_char == Some('\'');
+    let is_double_quoted = first_char == Some('"') && last_char == Some('"');
+
+    let unquoted = if first_char.is_some() && last_char.is_some() {
+        trimmed.trim_matches(|c| c == '\'' || c == '"')
+    } else {
+        trimmed
+    };
+    let suggested_name = generate_suggested_name(unquoted).unwrap();
+
+    if is_single_quoted {
+        format!("'{}'", suggested_name)
+    } else if is_double_quoted {
+        format!("\"{}\"", suggested_name)
+    } else {
+        suggested_name.to_string()
+    }
+}
+
+fn generate_suggested_name(member_name: &str) -> Option<String> {
+    match member_name {
+        "trimLeft" => Some("trimStart".to_string()),
+        "trimRight" => Some("trimEnd".to_string()),
```

Comment:
This function always allocate a new string. Don't need to do that, just return a `&str`